lib/serializer.ts

```diff
@@ -310,38 +311,58 @@ export class Serializer {
         }
       }
 
-      for (const key in modelProps) {
-        if (modelProps.hasOwnProperty(key)) {
-          const paths = this.splitSerializeName(modelProps[key].serializedName);
-          const propName = paths.pop();
+      for (const key of Object.keys(modelProps)) {
+        const propertyMapper = modelProps[key];
+
+        let propName: string | undefined;
+        let parentObject: any = payload;
+        if (this.isXML) {
+          if (propertyMapper.xmlIsWrapped) {
+            propName = propertyMapper.xmlName;
+          } else {
+            propName = propertyMapper.xmlElementName || propertyMapper.xmlName;
+          }
+        } else {
+          const paths = this.splitSerializeName(propertyMapper.serializedName);
```

Comment:
Good question--looks like Dan wondered the same thing. I think I just glossed over it because the way the property name is obtained for XML is different than in JSON. We could maybe just take the propName extracted on line 323, and pass it to splitSerializeName, and carry on from there.

Note that the C# runtime does not handle parameter flattening for XML and we aren't aware of any services that would need it for XML.